js/application.js

```diff
@@ -22,71 +24,147 @@ $(document).ready(function(){
   }
 
   var reset = function(){
-    for(i=1; i<=4; i++){
-      var resetGridRow="grid_"+i;
-      for (j=1; j<=4; j++){
-        var resetGridCol="column"+j;
-        $('#'+resetGridRow).children('#'+resetGridCol).attr('myAttr','canMove');
-        $('#'+resetGridRow).children('#'+resetGridCol).attr('combine','canCombine');
-      }
-    }
+    $('.grid').children().children().attr('myAttr', 'canMove');
+    $('.grid').children().children().attr('combine', 'canCombine');
   }
 
   var setColor = function(){
-    for(i=1; i<=4; i++){
+    for(i=1; i<=max; i++){
       var GridRow="grid_"+i;
-      for (j=1; j<=4; j++){
+      for (j=1; j<=max; j++){
         var GridCol="column"+j;
         var elementValue=$('#'+GridRow).children('#'+GridCol);
         var elementValueText=elementValue.text()
         switch(elementValueText){
+          case "0":
```

Comment:
A better way to do this is to use a hash.

For example,

```js
settings = {
  0: {
    background-color: '#CCC0B3',
    color: '#776e65',
    border: '8px solid #BBADA0'
  }
}
```

This way, you can write your code such that you can accomplish what you want to do in less than 5 lines, instead of a 100 line switch statements